es6/EntityFactory.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+import 'd2l-polymer-siren-behaviors/store/entity-store.js';
+
+// This creates and fetch a new entity. Whenever the entity changes onChange is called.
+export function entityFactory(entityType, href, token, onChange) {
+	const entityListener = new EntityListener();
+	const onChangeWrapped = (entity) => {
+		const entityWrapped = new entityType(entity, token, entityListener);
+		onChange(entityWrapped);
+	};
+
+	// This add the listener then calls the fetch.
+	entityListener.add(href, token, onChangeWrapped);
+}
+
+// Some times the entity doesn't exists so this allows the cleanup code to be cleaner.
+export function decompose(entity) {
```

Comment:
but why not `dispose()`, for 30% less zombie

Reply:
I totally forgot